### DISCUSSION TITLE
Are messages with rich `cards`, yet working on the Dashboard?

### DISCUSSION THREAD

**Discussion by AdsonCicilioti** [chatwoot/chatwoot#7999] [Q&A]
Hi guys!

Are messages with `cards` or interactive messages, yet working on the Dashboard?

#609 (#594) Isn't clear for me. The title says "...Widget and Dashboard".
Again... This is working on the Dashboard (Inboxes conversations)?

I would like to create incoming messages using rich metadata to generate this type of card via api:
 
![image](https://github.com/chatwoot/chatwoot/assets/6278398/ab4057b7-a4ff-44e5-b0a8-26ef225599aa)

_Originally posted by @AdsonCicilioti in https://github.com/chatwoot/chatwoot/discussions/7570#discussioncomment-6985315_

**Comment by pranavrajs** [comment]
@AdsonCicilioti Can you please share an example of such a card? Sending it to the customers makes sense. I couldn't figure out a use case where a customer can send such messages to the agents.

**Reply by AdsonCicilioti** [reply]
Hi @pranavrajs ! Thanks for your attention!

I recently worked on an application that has a personalized API that connects the operator's Whatsapp to Chatwoot, thus providing multi-service receiving messages from leads/customers that come from sponsored link campaigns such as those on Facebook and Instagram platforms.  These ads are specific to generating conversations on Whatsapp.
 With this scenario, when the customer clicks on the ad button, a message is already sent, whose object of the request includes, in addition to the text message, ad metadata, including link, title, description and thumbnail of the media.  This means that the message plus a card containing this ad metadata arrives on WhatsApp.
 The problem is that only the customer's message arrives in Chatwoot, and as a result, the attendants and the manager cannot know where the customer came from.
 It would be essential for Chatwoot to imitate Whatsapp's behavior.

 The same goes for messages that contain a link that contains rich metadata, so render a preview using that card.  Look:
![image](https://github.com/chatwoot/chatwoot/assets/6278398/b97d5780-aaf5-47b3-b086-19fcef31fdcb)

Perhaps supporting the rendering of links with rich metadata would already be a good basis for resolving these messages coming from this type of ad.

 What I did in this application was to create logic that identifies the metadata in the message request object and adds an attachment with the image simulating a card along with the message.  But I think this is far from ideal.  That's why I bring this question here.

 This link preview is present in several applications such as WhatsApp, Telegram, Messenger and others.

**Reply by WEBudoGT** [reply]
We're having the same issue when receiving whatsapp messages from Facebook AD campaign.
We're unable to get the card/image with the info of the ad clicked.
Are you guys planning to add this feature in future version @pranavrajs ?

**Reply by tuhinmallick** [reply]
Much needed feature
